cmd/k8s-operator/egress-services.go

```diff
@@ -136,9 +136,8 @@ func (esr *egressSvcsReconciler) Reconcile(ctx context.Context, req reconcile.Re
 	}
 
 	if !slices.Contains(svc.Finalizers, FinalizerName) {
-		l.Infof("configuring tailnet service") // logged exactly once
 		svc.Finalizers = append(svc.Finalizers, FinalizerName)
-		if err := esr.Update(ctx, svc); err != nil {
+		if err := esr.updateSvcSpec(ctx, svc); err != nil {
```

Comment:
We were at this point losing the status conditions that were not yet applied (because Update overwrites the object) so this was causing extra reconciles. Now it should be possible to make all spec and status updates in a single pass